bowling/src/ArgumentParser.cpp

```diff
@@ -0,0 +1,34 @@
+#include "ArgumentParser.hpp"
+
+#include <algorithm>
+
+ArgumentParser::ArgumentParser(int argc, char** argv) {
+    argc_ = argc;
+    argv_.reserve(argc_);
+    for (int i = 0; i <argc_ ; ++i) {
+        argv_.push_back(argv[i]);
+    }
+    appName_ = argv_[0];
+}
+
+std::string ArgumentParser::getInputDirectory() const {
+    if (argc_ < 2) {
+        return "";
+    }
+    return argv_[1]; 
+}
+
+std::string ArgumentParser::getOutputFileName() const {
+    if (argc_ < 3) {
+        return "";
+    }
+    return argv_[2]; 
```

Comment:
```suggestion
    if (argc_ <= 2) 
        return "";
    return argv_[2]; 
```
Duplicate code in `getInputDirectory`. You can extract method.

Reply:
I don't know if it's worth to extract a method here. Especially with your proposed changes and different comparison operators: `==` in `getInputDirectory()` and `<=` in `getOutputFileName()`. I'd rather stick with simple `if - return`  statement as it is.